src/GraphQL/Types/IGraphType.cs

```diff
@@ -40,7 +40,7 @@ public interface IProvideDeprecationReason
     /// <summary>
     /// Represents a graph type within the GraphQL schema.
     /// </summary>
-    public interface IGraphType : IProvideMetadata, IProvideDescription, IProvideDeprecationReason, INamedType
+    public interface IGraphType : IMetadataReader, IMetadataWriter, IProvideDescription, IProvideDeprecationReason, INamedType
```

Comment:
Note: `IGraphType` is a read/write interface (unfortunately).  So for example GraphQLAttribute implementations use `Modify(IGraphType)` and metadata needs to be writable.  Also the other inherited interfaces are writable such as `IProvideDescription`.